### DISCUSSION TITLE
Storybook v8 /w Next 14 - Why is my tsconfig not respected in storybook?

### DISCUSSION THREAD

**Discussion by Swoodend** [storybookjs/storybook#26895] [Help]
### Summary

I have a frontend app built using `NextJS v14`, `React 18`, and `@emotion/react v11` for CSS.

My storybook config is using the `@storybook/nextjs` framework. In my project I have a tsconfig file that specifies a few key details, namely:

```
{
  "compilerOptions": {,
    "jsxImportSource": "@emotion/react",
    "paths": {
      "@/*": ["./src/*"]
    },
  }
}
```

When I run storybook. I get an error saying the it cannot find anything imported from `@/some/path` because it doesn't know to resolve `@` to my `./src` directory as specified in the tsconfig, so I had to add something like this to the storybook.main:

```
  webpackFinal: async config => {
    config.resolve.alias['@'] = path.resolve(__dirname, '../src');
    return config;
  },
```

I also have to add the `/** @jsxImportSource @emotion/react */` pragma to any component where I want the breakpoints in emotion to work.

Why is this the case? Is storybook supposed to be reading my tsconfig file or not? Can I point storybook to my tsconfig so these configuration options can be picked up?

**Comment by valentinpalkovic** [comment]
The problem is that path aliases are only respected if the `baseUrl` option is set in your tsconfig file. Just set it to `.`. But I think we have already fixed this issue.

**Reply by valentinpalkovic** [reply]
This [fix](https://github.com/storybookjs/storybook/pull/26651) was released in 8.0.6

**Reply by Swoodend** [reply]
Thanks @valentinpalkovic. I'm using 8.0.5 so that makes sense.

Any ideas for why I have to specify the `/** @jsxImportSource @emotion/react */` pragma in my react components?

The commit for the MR you linked doesn't look like it would affect  why the `jsxImportSource` set in my tsconfig is not respected in storybook.

**Reply by valentinpalkovic** [reply]
You usually don’t have to. This [PR](https://github.com/storybookjs/storybook/pull/25203) has added support to set jsxImportSource in your tsconfig. Could you provide a reproduction?

**Reply by Swoodend** [reply]
I tried to create a reporduction to produce this bug and was unable to. I'm going to try a version bump and see what happens 🤔
